packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1933,12 +2010,29 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
 
   @override
   void didChangeNext(Route<dynamic>? nextRoute) {
+    if (nextRoute is ModalRoute<T> && nextRoute.delegatedTransition == null && nextRoute.subtreeContext == null && this.subtreeContext != null) {
+      nextRoute.setDelegatedTransition(this.subtreeContext!);
+    }
+    if (nextRoute is ModalRoute<T> && canTransitionTo(nextRoute) && nextRoute.delegatedTransition.runtimeType != this.delegatedTransition.runtimeType) {
```

Comment:
I wonder if there could ever be a case where you DO want to use the delegatedTransition for the same runtimeType? If so, we could have an overridable method on DelegatedTransition that defaults to this runtimeType check. But if we can't think of any case where you would need that then we shouldn't add it, just brainstorming.

Reply:
In those cases, I would think we'd want to push them towards making another class extending off of DelegatedTransition to get a different runtimeType. DelegatedTransition is fairly lightweight to make an extension off of.